myhomework4.php

```diff
@@ -0,0 +1,203 @@
+<?php
+
+require_once 'HttpStatusCodes.php';
+const MY_URI = "/api/checkLoginAndPassword";
+const MY_CONT_TYPE = "application/x-www-form-urlencoded";
+const FILE = "passwords";
+//function readHttpLikeInput()
+//{
+//    $f = fopen('php://stdin', 'r');
+//    $store = "";
+//    $toread = 0;
+//    while ($line = fgets($f)) {
+//        $store .= preg_replace("/\r/", "", $line);
+//        if (preg_match('/Content-Length: (\d+)/', $line, $m))
+//            $toread = $m[1] * 1;
+//        if ($line == "\r\n")
+//            break;
+//    }
+//    if ($toread > 0)
+//        $store .= fread($f, $toread);
+//    return $store;
+//}
+//
+//$contents = readHttpLikeInput();
+
+function outputHttpResponse($statuscode, $statusmessage, $headers, $body)
+{
+    $response = "HTTP/1.1 $statuscode $statusmessage" . PHP_EOL;
+    $response .= "Date: " . date("l, d F  Y h:i:sa") . PHP_EOL;
+    $response .= "Server: Apache/2.2.14 (Win32)" . PHP_EOL;
+    $response .= "Content-Length: " . strlen($body) . PHP_EOL;
+    $response .= "Connection: Closed" . PHP_EOL;
+    $response .= "Content-Type: text/html; charset=utf-8" . PHP_EOL;
+    $response .= PHP_EOL . "$body";
+
+    echo $response;
+}
+
+function processHttpRequest($method, $uri, $headers, $body)
+{
+    try{
+        $statusCode = getStatusCode($headers, $uri, $body);
```

Comment:
simplify this behaviour. you overcomplicated the logic for this simple thing

Reply:
I don't understand what exactly I need to simplify